x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx

```diff
@@ -223,6 +214,26 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
   const machineLearningJobId = formMachineLearningJobId ?? initialState.machineLearningJobId;
   const queryBar = formQuery ?? initialState.queryBar;
 
+  const setRuleTypeCallback = useSetFieldValueWithCallback('ruleType', setFieldValue, ruleType);
+
+  const handleSetRuleFromTimeline = useCallback(
+    ({ index: timelineIndex, queryBar: timelineQueryBar }) => {
+      const setQuery = () => {
+        setFieldValue('index', timelineIndex);
+        setFieldValue('queryBar', timelineQueryBar);
+      };
+      if (timelineQueryBar.query.language === 'eql') {
+        setRuleTypeCallback('eql', setQuery);
```

Comment:
`setRuleTypeCallback` sets the `ruleType` field value to `'eql'`, and once that completes invokes the callback `setQuery`